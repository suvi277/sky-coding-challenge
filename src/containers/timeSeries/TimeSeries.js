import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchTimeSeries } from 'actions';
import Loader from 'components/loader/Loader';
import Chart from 'components/chart/Chart';
import Aux from 'components/hoc/Aux';
import moment from 'moment';

class TimeSeries extends Component {
	componentDidMount() {
		this.props.fetchTimeSeries();
		this.prepareChartData();
	}

	prepareChartData() {
		const { data } = this.props.timeSeries;
		let day;

		let labels = '',
			lineData = '';

		if (data.length) {
			lineData = data.map((time) => time.value);
			labels = data.map((time) => {
				const date = moment(new Date(time.timestamp));
				day = date.format('LL');
				return date.format('mm:ss a');
			});
		}

		return { labels, lineData, day };
	}

	render() {
		const { isLoaded } = this.props.timeSeries;
		return (
			<Aux>
				{!isLoaded && <Loader />}
				{isLoaded && <Chart {...this.prepareChartData()} />}
			</Aux>
		);
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		fetchTimeSeries: () => dispatch(fetchTimeSeries())
	};
};

export default connect(({ timeSeries }) => ({ timeSeries }), mapDispatchToProps)(TimeSeries);
