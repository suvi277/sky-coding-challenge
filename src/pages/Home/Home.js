import React from 'react';
import MainLayout from 'components/hoc/Layout';
import PageHeading from 'components/pageHeading/PageHeading';
import TimeSeries from 'containers/timeSeries/TimeSeries';

const Home = () => {
	return (
		<MainLayout>
			<PageHeading title="Time Series Dashboard" />
			<div className="pt-4">
				<TimeSeries />
			</div>
		</MainLayout>
	);
};
export default Home;
