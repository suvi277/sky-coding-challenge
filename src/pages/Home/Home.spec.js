import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';
import TimeSeries from 'containers/timeSeries/TimeSeries';

describe('Home Component', () => {
	const wrapper = shallow(<Home />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should exist MainLayout Component', () => {
		expect(wrapper.find('MainLayout')).toHaveLength(1);
	});

	it('should exist PageHeading Component', () => {
		expect(wrapper.find('PageHeading')).toHaveLength(1);
	});

	it('should exist TimeSeries Component', () => {
		expect(wrapper.exists(TimeSeries)).toBeTruthy();
	});
});
