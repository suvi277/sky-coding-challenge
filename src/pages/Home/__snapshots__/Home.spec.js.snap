// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders as expected 1`] = `
<MainLayout>
  <PageHeading
    title="Time Series Dashboard"
  />
  <div
    className="pt-4"
  >
    <Connect(TimeSeries) />
  </div>
</MainLayout>
`;
